numpy/lib/function_base.py

```diff
@@ -1801,6 +1801,10 @@ def place(arr, mask, vals):
            [44, 55, 44]])
 
     """
+    if not isinstance(arr, np.ndarray):
```

Comment:
See #7003, which does a complete revamp of ```arr_insert``` (that is tied to ```_insert```) and does the type checking.  In the case of ```np.place```, this PR should serve as a stopgap measure until #7003 can be merged.